src/main/java/com/iridium/iridiumteams/commands/BlockValueCommand.java

```diff
@@ -0,0 +1,107 @@
+package com.iridium.iridiumteams.commands;
+
+import com.iridium.iridiumcore.utils.StringUtils;
+import com.iridium.iridiumteams.IridiumTeams;
+import com.iridium.iridiumteams.SettingType;
+import com.iridium.iridiumteams.configs.inventories.BlockValuesTypeSelectorInventoryConfig;
+import com.iridium.iridiumteams.database.IridiumUser;
+import com.iridium.iridiumteams.database.Team;
+import com.iridium.iridiumteams.database.TeamSetting;
+
+import com.iridium.iridiumteams.gui.BlockValueGUI;
+import com.iridium.iridiumteams.gui.BlockValuesTypeSelectorGUI;
+import lombok.NoArgsConstructor;
+import org.bukkit.Bukkit;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@NoArgsConstructor
+public class BlockValueCommand<T extends Team, U extends IridiumUser<T>> extends Command<T, U> {
+
+    public BlockValueCommand(List<String> args, String description, String syntax, String permission, long cooldownInSeconds) {
+        super(args, description, syntax, permission, cooldownInSeconds);
+    }
+
+    @Override
+    public boolean execute(U user, String[] args, IridiumTeams<T, U> iridiumTeams) {
+
+        Player player = user.getPlayer();
+        Optional<T> team;
+
+        BlockValuesTypeSelectorInventoryConfig blockValuesTypeSelectorInventoryConfig = iridiumTeams.getInventories().blockValuesTypeSelectorGUI;
+
+        String teamArg;
+        if (args.length > 1) {
+            teamArg = args[1];
+        } else {
+            teamArg = player.getName();
+        }
+
+        team = iridiumTeams.getTeamManager().getTeamViaNameOrPlayer(teamArg);
+
+        if (!team.isPresent() && args.length > 1) {
+            player.sendMessage(StringUtils.color(iridiumTeams.getMessages().teamDoesntExistByName.replace("%prefix%", iridiumTeams.getConfiguration().prefix)));
+            return false;
+        }
+
+        if (!team.isPresent()) {
+            player.sendMessage(StringUtils.color(iridiumTeams.getMessages().dontHaveTeam.replace("%prefix%", iridiumTeams.getConfiguration().prefix)));
+            return false;
+        }
+
+        TeamSetting teamSetting = iridiumTeams.getTeamManager().getTeamSetting(team.get(), SettingType.VALUE_VISIBILITY.getSettingKey());
+
+        if (teamSetting == null) {
+            player.sendMessage(StringUtils.color(iridiumTeams.getMessages().invalidPermission.replace("%prefix%", iridiumTeams.getConfiguration().prefix)));
+            return false;
+        }
+
+        if (teamSetting.getValue().equalsIgnoreCase("Private") && !iridiumTeams.getTeamManager().getTeamMembers(team.get()).contains(user) && !user.isBypassing()) {
+            player.sendMessage(StringUtils.color(iridiumTeams.getMessages().teamIsPrivate.replace("%prefix%", iridiumTeams.getConfiguration().prefix)));
+            return false;
+        }
```

Comment:
```suggestion
        if (teamSetting != null && teamSetting.getValue().equalsIgnoreCase("Private") && !iridiumTeams.getTeamManager().getTeamMembers(team.get()).contains(user) && !user.isBypassing()) {
            player.sendMessage(StringUtils.color(iridiumTeams.getMessages().teamIsPrivate.replace("%prefix%", iridiumTeams.getConfiguration().prefix)));
            return false;
        }
```

`invalidPermission` is used when we do /is permission command, I dont think we should be sending this here since users will most likely think its a bug when infact we have just disabled the permission in the config.

Instead I think if we have disabled it in the config, we should treat it as though the setting doesnt exist and do whatever is the "Default" which in this case I think should be Public

Reply:
i thought i fixed this